increment ATMOS_SHLVL each time atmos terraform shell is called (#803)
cloudposse/atmos: website/docs/cli/commands/terraform/terraform-shell.mdx

- Inside the shell, the user can execute all `terraform` commands using the native syntax

> - Sets the ATMOS_SHLVL environment variable to track shell nesting levels.

Reviewer: ```suggestion
- Sets the `ATMOS_SHLVL` environment variable to track shell nesting levels.
```

    - If ATMOS_SHLVL is not set, it initializes to 1.

>     - If ATMOS_SHLVL is already set, it increments the value by 1 for each new shell.

Reviewer: ```suggestion
    - If `ATMOS_SHLVL` is already set, it increments the value by 1 for each new shell.
```
```suggestion
    - If ATMOS_SHLVL is already set, it increments the value by 1 for each new shell.
```

> - Sets the `ATMOS_SHLVL` environment variable to track shell nesting levels.
>     - If `ATMOS_SHLVL` is not set, it initializes to 1.
>     - If `ATMOS_SHLVL` is already set, it increments the value by 1 for each new shell.
> 

Reviewer: ```suggestion
- Atmos sets the `ATMOS_SHLVL` environment variable to track the nesting level of shells:
    - If `ATMOS_SHLVL` is not already set, Atmos initializes it to `1`.
    - If `ATMOS_SHLVL` is already set, Atmos increments its value by `1` for each new nested shell.

```
